docs/doxygen/structure_review.md

```diff
@@ -1,42 +1,54 @@
-2. DPBuffer will be split into multiple types. For example, UVW will be a buffer on its own. Using a vector as input for the process function, the DPSteps have access to the information they need and nothing more or less. It allows any kind of data type to be streamed in DP3.
+1. To solve problem 1, a possible solution is that the process function should no longer have a single DPBuffer as input but rather a vector of DPBuffers. DP3 should orchestrate the process and make sure that the data flows as described.
+2. For solving problem 2, we could introduce different DPBuffer types. For example, a buffer then only holds UVW data and nothing else, or a buffer for solutions. Using a vector as input for the process function, the DPSteps have access to the information they need and nothing more or less. DP3 then supports streaming any kind of data type.
+3. Problem 3 can be solved by providing the user with a syntax to define with model is to be applied per direction. DDECal will then initialize the models and applies it to said directions.
+
+We will focus on solving problem 3, since problems 1 and 2 have workarounds.
 
 ## Implementation plan
 During the review sessions we concluded that to enable multiple predict types in DDECal not much needs to be done. Therefore, we have decided to not refactor the DP3 structure significantly because we can implement the required feature without changing the core structure of DP3.
 
-We have identified 2 key refactors that fulfil the required use cases:
-1. Refactor the creation of ApplyCal steps in Prediction steps. DDECal should be responsible for the creation of ApplyCal, not Predict. This should therefore be done in the DDECal step and note in the Predict step.
-2. Factor out IDGPredict as separate step so that it can be used outside of DDECal.
-3. Refactor the nested DDECal model pipeline so that it can be defined separately for every direction instead of one to rule them all. The metadata must not change in any of the nested model steps of DDECal. This requires a refactors in the process function of DDECal.cc.
+We have identified four key refactors that fulfil the required use cases:
+1. Refactor the creation of ApplyCal steps inside Predict steps. Currently, the DDECal step creates Predict substeps which in turn create ApplyCal subsubsteps. DDECal should be responsible for the creation of ApplyCal, not Predict, so the ApplyCal steps become substeps of DDECal instead of being subsubsteps inside Predict.
+2. Currently, IDGPredict is implemented using a custom interface. We should factor out this code and convert it to the generic DPStep interface, so it can also be used outside of DDECal.
+3. Refactor the nested DDECal model pipeline so that it can be defined separately for every direction instead of one to rule them all. The metadata must not change in any of the nested model steps of DDECal. This requires refactoring the steps setup in the constructor of DDECal.cc.
```

Comment:
```suggestion
3. Refactor the nested DDECal model pipeline so that it can be defined separately for every direction instead of one to rule them all. The metadata must not change in any of the nested model steps of DDECal. This requires refactoring the steps setup in the constructor of DDECal.
```